solarpilot/Heliostat.cpp

```diff
@@ -745,23 +745,22 @@ Reflector *Heliostat::getPanelById(int id){
 
 	for (int j=0; j<(int)ncantx; j++) {
 		for (int i=0; i<(int)ncanty; i++) {
-			if (_panels[j,i].getId() == id){
-				return &_panels[j,i];
```

Comment:
Seems like a fairly insidious bug, from the perspective that an operator [] with two operands is not defined for a matrix_t, and so only the right operand gets evaluated according to https://stackoverflow.com/questions/54142/how-does-the-comma-operator-work .

Reply:
I agree, nice find!